packages/react-components/src/viewport-controls/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useMemo } from 'react';
+import { useViewport } from '../viewport-manager';
+import { DateRangePicker, DateRangePickerProps, NonCancelableCustomEvent } from '@awsui/components-react';
+import React from 'react';
+
+const mapUnit = (unit: DateRangePickerProps.TimeUnit) => {
+  switch (unit) {
+    case 'day':
+      return 'd';
+    case 'minute':
+      return 'm';
+    case 'hour':
+      return 'h';
+    case 'week':
+      return 'w';
+  }
+};
+
+const reverseMapUnit = (symbol: string): DateRangePickerProps.TimeUnit => {
+  switch (symbol) {
+    case 'd':
+      return 'day';
+    case 'h':
+      return 'hour';
+    case 'w':
+      return 'week';
+    default:
+      return 'minute';
+  }
+};
+
+const parseDuration = (duration: string) => {
+  const amount = parseInt(duration);
+  const unitRef = duration.split(`${amount}`)[1];
+
+  return {
+    amount,
+    unit: reverseMapUnit(unitRef),
+  };
+};
+
+const ViewportControls = ({ value: defaultValue, onChange, ...props }: DateRangePickerProps) => {
+  const { update, reset, viewport } = useViewport();
+
+  const value: DateRangePickerProps.Value | null = useMemo(() => {
+    const relativeViewport = viewport as { duration: string };
```

Comment:
nit: this is interesting but maybe we could achieve the same with predicates? 

So instead of 
```
if (relativeViewport?.duration) {
```
something like
```
isMinimalStaticViewport = (viewport: MinimalViewPortConfig): viewport is MinimalLiveViewport =>
  'duration' in viewport;

if (isRelativeViewport(viewport)) {
```